src/FocusApp.Client/Views/Social/UserBadgesPage.cs

```diff
@@ -0,0 +1,235 @@
+            ColumnDefinitions = Columns.Define(Star, Star),
+            BackgroundColor = AppStyles.Palette.DarkMauve,
+
+            Children =
+            {
+            // Header
+            new Label
+            {
+                Text = "Badges",
+                TextColor = Colors.Black,
+                FontSize = 40
+            }
+            .Row(0)
+            .Column(0)
+            .Center()
+            .ColumnSpan(2)
+            .CenterVertical()
+            .Paddings(top: 5, bottom: 5, left: 5, right: 5),
+
+            // Back Button
+            new Button
+            {
+                Text = SolidIcons.ChevronLeft,
+                TextColor = Colors.Black,
+                FontFamily = nameof(SolidIcons),
+                FontSize = 40,
+                BackgroundColor = Colors.Transparent
+            }
+            .Left()
+            .CenterVertical()
+            .Paddings(top: 10, bottom: 10, left: 15, right: 15)
+            .Column(0)
+            .Invoke(button => button.Released += (sender, eventArgs) =>
+            BackButtonClicked(sender, eventArgs)),
+
+            // Header & Content Divider
+            new BoxView
+            {
+                Color = Color.Parse("Black"),
+                WidthRequest = 400,
+                HeightRequest = 2,
+            }
+            .Bottom()
+            .Row(0)
+            .Column(0)
+            .ColumnSpan(5),
+
+            new ScrollView
+            {
+               Content = _flexLayout
+            }
+            // Badges FlexLayout
+            .ColumnSpan(2)
+            .Row(2)
+            }
+        };
+
+    }
+
+    protected override async void OnAppearing()
+    {
+        base.OnAppearing();
+
+        // Check if the user is logged in
+        if (_authenticationService.CurrentUser == null)
+        {
+            var loginPopup = (EnsureLoginPopupInterface)_popupService.ShowAndGetPopup<EnsureLoginPopupInterface>();
+            loginPopup.OriginPage = nameof(ShopPage);
+            return;
+        }
+
+        // Fetch badges from the local database and display them
+        var localBadges = await FetchBadgesFromLocalDb();
+        var userBadges = await FetchBadgesFromAPI();
+        DisplayBadges(localBadges, userBadges);
+    }
+
+    private async Task<List<Badge>> FetchBadgesFromLocalDb()
+    {
+        try
+        {
+            // Fetch badges from the local database
+            return await _localContext.Badges.ToListAsync();
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error when fetching badges from local DB.", ex);
+        }
+    }
+
+
+    private async Task<List<Guid>> FetchBadgesFromAPI()
+    {
+        try
+        {
+            return await _localContext.UserBadges?.Select(ub => ub.BadgeId).ToListAsync();
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error when fetching badges from API.", ex);
+        }
+    }
```

Comment:
Could we rename this method to describe what it's doing? Also since this doesn't have another try-catch wrapped around it, the exception might crash the app. Instead could we log the exception